app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.kt

```diff
@@ -112,11 +115,15 @@ interface CommentsViewModel {
                 .share()
 
             commentEnvelope
-                .map { it.comments }
                 .filter { ObjectUtils.isNotNull(it) }
                 .compose(bindToLifecycle())
                 .subscribe {
-                    commentsList.onNext(it)
+                    it.totalCount?.let { count ->
+                        if (count > 0) {
```

Comment:
you could refactor these lines to be 
`this.setEmptyState.onNext(count > 0)`
so you don't need the if condition inside the let block

Reply:
great. thanks